File: sample_repos/java/SED/3/src/main/java/ic/doc/templatemethod/FibonacciSequence.java
package ic.doc.templatemethod;

public class FibonacciSequence extends Sequence {

    @Override
    public int generate_term(int i) {
        if (i < 2) {
            return 1;
        }
        return generate_term(i - 1) + generate_term(i - 2);
    }
}

File: sed_repos/python/SED3/src/template_method/fibonacci_sequence.py
from src.template_method.sequence import Sequence


class FibonacciSequence(Sequence):
    def generate_term(self, i: int) -> int:
        if i < 2:
            return 1
        return self.generate_term(i - 1) + self.generate_term(i - 2)
